packages/sdk/src/StreamrClient.ts

```diff
@@ -426,12 +440,16 @@ export class StreamrClient {
         permissionFilter: SearchStreamsPermissionFilter | undefined,
         orderBy: SearchStreamsOrderBy = { field: 'id', direction: 'asc' }
     ): AsyncIterable<Stream> {
-        return this.streamRegistry.searchStreams(
+        logger.debug('Search for streams', { term, permissionFilter })
+        if ((term === undefined) && (permissionFilter === undefined)) {
+            throw new Error('Requires a search term or a permission filter')
+        }
```

Comment:
Is this a new limitation?

Reply:
No. Moved from `searchStream.ts#47` (this is the only caller). Good to validate user input in the API method, i.e. in the `StreamrClient` instead of `searchStreams` utility.